### PR TITLE
feat(docker): refactor dockerfile + bootstrapping

### PR DESCRIPTION
**Pull request by serge1peshcoff**
1) moved all the dependencies to `package.json`
2) removed all the custom workarounds/bootstrapping, now it's just `npm install` (as it's a node app)
3) removed all bootstrapping scripts (makefile/start-dev.sh) as we now can do it with `./helper.sh --something hubot`

caveats: we need to rebuild the container if the deps list was changed, but this shouldn't happen often right?

### CODE REVIEW

**Comment by linuxbandit** [comment]
and it just works?

**Comment by serge1peshcoff** [comment]
It should in theory. From what I've tested, the container is building, the app is starting with no errors, but I couldn't make it react to my slack messages. This should be something with my setup though

**Comment by linuxbandit** [comment]
then it doesn't just work :D I don't remember the rationale of doing it this way when it was already working and it was following all the 'standards'/guides (building through yeoman etc)?

**Comment by serge1peshcoff** [comment]
point is, we can do it once when creating a project, and we can also do it every time we start/build the container, while we won't need it most of the time.
Anyway, I won't merge this until I can make it work locally

**Comment by serge1peshcoff** [comment]
yep, it works for me now

**Comment by linuxbandit** [comment]
i will test it too, Idk it doesn't convince me doing different than all the setup guides

File: docker/Dockerfile [hunk 1]
```diff
@@ -1,43 +1,11 @@
-FROM node:14.2 AS builder
```
Comment (linuxbandit) [inline docker/Dockerfile:1]: better use `-alpine`

File: docker/Dockerfile [hunk 2]
```diff
@@ -1,43 +1,11 @@
...
-
-RUN jq --arg HUBOT_VERSION "$HUBOT_VERSION" '.dependencies.hubot = $HUBOT_VERSION' package.json > /tmp/package.json \
-      && mv /tmp/package.json .
-
-FROM node:14.2-alpine AS prod
-
-RUN addgroup -S hubot && adduser -S hubot -G hubot \
-      && apk update && apk add --no-cache jq
-
-USER hubot
```
Comment (linuxbandit) [inline docker/Dockerfile:33]: also run as unprivileged

File: docker/Dockerfile [hunk 3]
```diff
@@ -1,43 +1,11 @@
...
-COPY --from=builder --chown=hubot /home/hubot/ /home/hubot/
-COPY --chown=hubot ./scripts /home/hubot/scripts
-COPY --chown=hubot ./utils /home/hubot/utils
-COPY --chown=hubot ./docker/entrypoint.sh /home/hubot/entrypoint.sh
-COPY --chown=hubot ./external-scripts.json /home/hubot/external-scripts.json
+RUN npm install
 
-ENTRYPOINT ["/home/hubot/entrypoint.sh"]
\ No newline at end of file
+CMD npx hubot -a slack
```
Comment (linuxbandit) [inline docker/Dockerfile:11]: use ENTRYPOINT, not CMD

**Review by linuxbandit — CHANGES_REQUESTED** [review]
also imho the name `hubot-slack-docker` is more specific, which is good

**Comment by serge1peshcoff** [comment]
@linuxbandit can you check once more?

**Review by linuxbandit — APPROVED** [review]
Still skeptic about the different building system (+ this way if we add a plugin we have to rebuild instead of simply relaunching the container, but that is minor). Buuut if it works for you, be it

**Pull request merged**